pkg/build/webhook/generic/generic.go

```diff
@@ -52,7 +42,7 @@ func (p *WebHookPlugin) Extract(buildCfg *api.BuildConfig, secret, path string,
 		if len(body) == 0 {
 			return revision, true, nil
 		}
-		var data webHookEvent
+		var data api.GenericWebHookEvent
 		if err = json.Unmarshal(body, &data); err != nil {
```

Comment:
In the next version we really need to be using the encode decode libraries, with conversion, and add TypeMeta to the event.